### PR TITLE
[StateService] p2p

### PR DESCRIPTION
**Pull request by ZhangTao1596**
Use p2p to verify

* Add `network` part to define all data type in state service`ExtensiblePayload`.
* Use `ExtensiblePayload` to send signature via p2p.
* Remove vote rpc

### CODE REVIEW

**Comment by ProDog** [comment]
Tested OK.
```
[16:27:00.123] new validate process, height=316, index=0, ongoing=1
[16:27:03.125] relay vote, height=316, retry=0
[16:27:03.148] vote received, height=316, index=3
[16:27:03.853] vote received, height=316, index=1
[16:27:03.861] relay state root, height=316, root=0xd6221c54f97f3a1e021929b9744dfbcec6d4c85f42d1907660551d1a213770ea
[16:27:03.868] persisted state root, height=316
[16:27:09.207] new validate process, height=317, index=0, ongoing=1
[16:27:12.215] relay vote, height=317, retry=0
[16:27:12.236] vote received, height=317, index=3
[16:27:13.129] vote received, height=317, index=1
[16:27:13.150] relay state root, height=317, root=0xb0e2450b08ab242edff9b3ecfd3089bd43695841877da8967ba9ffb91b872ea8
[16:27:13.171] persisted state root, height=317
[16:27:20.217] new validate process, height=318, index=0, ongoing=1
[16:27:23.225] relay vote, height=318, retry=0
[16:27:23.238] vote received, height=318, index=3
[16:27:23.243] vote received, height=318, index=1
[16:27:23.281] relay state root, height=318, root=0x244b718ea87ada64d7d5c0fe2c9e7a179dc3b67c5761cabcf926439dc5e23db9
[16:27:23.300] persisted state root, height=318
[16:27:30.268] new validate process, height=319, index=0, ongoing=1
[16:27:33.278] relay vote, height=319, retry=0
[16:27:33.286] vote received, height=319, index=3
```

```
neo> get proof 0x00682c9258176b8118a3782ad549876f8a13290f4717d33e83fd257eb2c39415 0x79bcd398505eb779df6e67e4be6c14cded08e2f2 Fw==
Bfv///8XBiQBAQ8DEpeidtbgxHDaMyv4r95RtBA1hB7hGdccpEVGs9R3pi7yAAQEBAQEBAQEA7l84HFtRI5V11s58vA+8CZ5GArFLkGUYLO98RLaMaYmA5MEnx0upnVI45XTpoUDRvwrlPD59uWy9aIrdS4T0D2cA6Rwv/l3GmrctRzL1me+iTUFdDgooaz+esFHFXJdDANfA2bdshZMp5ox2goVAOMjvoxNIWWOqjJoRPu6ZOw2kdj6AzxcjJcMbMHoCvRfr4rEqyd5NPWiZ8OhoL2t105v3gnvA1nf4WdI+jksYz0EJgzBukK8rEzz8jE2cb2Zx2fytVyQBANC7v2RaLMCRF1XgLpSri12L2IwL9Zcjz5LZiaB5nHKNgQpAQYPDw8PDw8DggFffnsVMyqAfZjg+4gu97N/gKpOsAK8Q27s56tijRlSAAMm26DYxOdf/IjEgkE/u/CoRL6dDnzvs1dxCg/00esMvgPGioeOqQCkDOTfliOnCxYjbY/0XvVUOXkceuDm1W0FzQQEBAQEBAQEBAQEBAQEBJIABAPH1PnX/P8NOgV4KHnogwD7xIsD8KvNhkTcDxgCo7Ec6gPQs1zD4igSJB4M9jTREq+7lQ5PbTH/6d138yUVvtM8bQP9Df1kh7asXrYjZolKhLcQ1NoClQgEzbcJfYkCHXv6DQQEBAOUw9zNl/7FJrWD7rCv0mbOoy6nLlHWiWuyGsA12ohRuAQEBAQEBAQEBAYCBAIAAgA=

neo> verify proof 0x00682c9258176b8118a3782ad549876f8a13290f4717d33e83fd257eb2c39415 Bfv///8XBiQBAQ8DEpeidtbgxHDaMyv4r95RtBA1hB7hGdccpEVGs9R3pi7yAAQEBAQEBAQEA7l84HFtRI5V11s58vA+8CZ5GArFLkGUYLO98RLaMaYmA5MEnx0upnVI45XTpoUDRvwrlPD59uWy9aIrdS4T0D2cA6Rwv/l3GmrctRzL1me+iTUFdDgooaz+esFHFXJdDANfA2bdshZMp5ox2goVAOMjvoxNIWWOqjJoRPu6ZOw2kdj6AzxcjJcMbMHoCvRfr4rEqyd5NPWiZ8OhoL2t105v3gnvA1nf4WdI+jksYz0EJgzBukK8rEzz8jE2cb2Zx2fytVyQBANC7v2RaLMCRF1XgLpSri12L2IwL9Zcjz5LZiaB5nHKNgQpAQYPDw8PDw8DggFffnsVMyqAfZjg+4gu97N/gKpOsAK8Q27s56tijRlSAAMm26DYxOdf/IjEgkE/u/CoRL6dDnzvs1dxCg/00esMvgPGioeOqQCkDOTfliOnCxYjbY/0XvVUOXkceuDm1W0FzQQEBAQEBAQEBAQEBAQEBJIABAPH1PnX/P8NOgV4KHnogwD7xIsD8KvNhkTcDxgCo7Ec6gPQs1zD4igSJB4M9jTREq+7lQ5PbTH/6d138yUVvtM8bQP9Df1kh7asXrYjZolKhLcQ1NoClQgEzbcJfYkCHXv6DQQEBAOUw9zNl/7FJrWD7rCv0mbOoy6nLlHWiWuyGsA12ohRuAQEBAQEBAQEBAYCBAIAAgA=
AAI=
neo>
neo> state height
LocalRootIndex: 326, ValidatedRootIndex: 325

neo> state root 326
{"version":0,"index":326,"roothash":"0xd073aca7023f924bbfcf16111a35d1aec16e318563eb821cdf6e7a8411eac1d5","witness":{"invocation":"DECksGFd2scftoLraX9OJyGQIZk3swwRqER1DPm1HQzKTu81A7ERgw6zRNQVUNs75hzZj11coqnAPD9UgvyRpbdIDECnf4emaNAiAmkgf3mK3TcEJWtgeW/NW40kU0lk5VeXeQHi3YKetoRupC0JAr/mj8W5HHEBlCaR9l5ch1iKu9CqDEB4krZiAZkLEnqaKynFooYsX/j0XZutQJXY87xgEqxaRUE/NlleyyLEVkHU/CgU7PLH\u002BGIhaZOcEf2h35LRvZna","verification":"EwwhAwAqLhjDnN7Qb8Yd2UoHuOnz\u002BgNqcFvu\u002BHZCUpVOgtDXDCECAM1gQDlYokm5qzKbbAjI/955zDMJc2eji/a1GIEJU2EMIQKXhyDsbFxYdeA0d\u002BFsbZj5AQhamA13R64ysGgh19j6UwwhA8klCeQozdf3pP3UqXxniRC0DxRl3d5PBJ9zJa8zgHkpFAtBE43vrw=="}}
neo> 
```

**Review by shargon — DISMISSED** [review]

File: src/StateService/Network/StateMessage.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+using Neo.IO;
+using System;
+using System.IO;
+
+namespace Neo.Plugins.StateService.Network
+{
+    public enum MessageType : byte
```
Comment (erikzhang) [inline src/StateService/Network/StateMessage.cs:7]: One file one class.

Comment (ZhangTao1596) [reply src/StateService/Network/StateMessage.cs]: Done.

File: src/StateService/Network/StateRoot.cs [hunk 2]
```diff
@@ -10,9 +10,9 @@
 using System;
 using System.IO;
 
-namespace Neo.Plugins.StateService.Storage
+namespace Neo.Plugins.StateService.Network
 {
-    public class StateRoot : IVerifiable, ISerializable
+    class StateRoot : StateMessage, IVerifiable
```
Comment (erikzhang) [inline src/StateService/Network/StateRoot.cs:15]: Why not send this message directly? I think we don't need to wrap it in an `ExtensiblePayload`, right?

Comment (ZhangTao1596) [reply src/StateService/Network/StateRoot.cs:15]: Node without will relay `ExtensiblePayload` but doesn't relay StateRoot.

Comment (erikzhang) [reply src/StateService/Network/StateRoot.cs:15]: Then it shouldn't implement `IVerifiable`. We should verify the `ExtensiblePayload`.

Comment (ZhangTao1596) [reply src/StateService/Network/StateRoot.cs:15]: StateRoot has the multi-sig of verifiers ExtensiblePayload doesn't.

Comment (erikzhang) [reply src/StateService/Network/StateRoot.cs:15]: I don't think so. All witness support multiple signatures.

Comment (ZhangTao1596) [reply src/StateService/Network/StateRoot.cs:15]: Node with stateservice only need to store `version+index+roothash+witness` so verifiers only sign `version+index+roothash` when exchange signatures.
If we put muti-sig in `ExtensiblePayload`, node need to store this payload. We don't need to `Category `, `ValidBlockStart`, `ValidBlockEnd`.
Say if neo p2p protocol doesn't support `Block` message, will you put block in `Data` field of `ExtensiblePayload` or  use `ExtansiblePayload` as block?

**Review by erikzhang — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**